okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java

```diff
@@ -132,6 +135,34 @@ public AndroidPlatform(Class<?> sslParametersClass, OptionalMethod<Socket> setUs
     }
   }
 
+  @Override public Object captureAllocationSite() {
+    if (missingCloseGuard.get()) return null;
+
+    try {
+      Class<?> closeGuardClass = Class.forName("dalvik.system.CloseGuard");
```

Comment:
Yeah, sigh. I like this approach ’cause it’s likely to do the right thing in development vs. production. It’s still a bit of a hack, but defensible.